### PR TITLE
Add tokens

### PR DESCRIPTION
**Pull request by jdbann**
- Adds tokens which belong to maps
- Tokens positions match the map as it is zoomed and moved
- Tokens can be moved and locations are updated in real-time for other users

### CODE REVIEW

**Comment by jdbann** [comment]
Sharing early as this will be affected by your background image investigation @cpytel

**Comment by jdbann** [comment]
And also because I wanted to play 😂 https://table-add-tokens-a3w7mxweemdsp.herokuapp.com/campaigns/d5a52490-db1f-47a0-8d77-2cc3dcc227c8

**Comment by jdbann** [comment]
@cpytel Updated for panning and zooming. Would appreciate your thoughts

**Comment by cpytel** [comment]
@jdbann it looks to me like the sizing of the tokens relative to the zoom might be the opposite of what it should be?

Here I am zoomed out, and the tokens are big:

Screen Shot 2020-04-04 at 1 24 56 PM

And here I am zoomed in and the tokens get smaller as I zoom in:

Screen Shot 2020-04-04 at 1 24 37 PM

**Comment by jdbann** [comment]
Hmm... I haven’t put anything in to resize them. They’re also positioned with 0, 0 as the top left, when the zooming would be better if that was the center of the token. I’ll add issues for both.

**Comment by cpytel** [comment]
Oh, I see, it's just an optical illusion and the tokens aren't actually different sizes.

File: spec/system/manage_maps_spec.rb [hunk 1]
```diff
@@ -51,8 +51,8 @@
     find(".campaign-map-selector", text: "Dwarven Excavation").click
     click_and_move_map(map, from: { x: 300, y: 300 }, to: { x: 50, y: 50 })
 
-    expect(page).to have_map_with_data(map, "x", "500")
-    expect(page).to have_map_with_data(map, "y", "500")
+    expect(page).to have_map_with_data(map, "x", "550")
```
Comment (cpytel) [inline spec/system/manage_maps_spec.rb:54]: I'm not sure why this changed?

File: app/models/token.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class Token < ApplicationRecord
+  belongs_to :map
+  has_one_attached :image
+
+  validates :name, presence: true
+  validates :image, presence: true
+
+  def scaled_x
```
Comment (cpytel) [inline app/models/token.rb:8]: Unless I'm missing it, these two functions aren't used?

File: spec/system/manage_maps_spec.rb (hunk 1 above)
Comment (jdbann) [reply spec/system/manage_maps_spec.rb:54]: I had to ensure that maps created by the factories were centered: https://github.com/jdbann/table/pull/7/files#diff-b709bad60390eff76d8e893b0e31911dR18

File: app/models/token.rb (hunk 2 above)
Comment (jdbann) [reply app/models/token.rb:8]: Ah yes, I was trying to position them correctly in the partial but scrapped it.

Comment (jdbann) [reply app/models/token.rb]: Removed in c0ebd3a17a66c5acb6b6379a0c849bd28d9329e8

**Pull request merged**